packages/core/test/unit/Site.test.js

```diff
@@ -276,123 +276,237 @@ test('Site should not deploy without a built site', async () => {
-
-    const json = {
-      ...PAGE_NJK,
-      'site.json': SITE_JSON_DEFAULT,
-      _site: {},
-    };
-    fs.vol.fromJSON(json, '');
-    const site = new Site('./', '_site');
-    await site.deploy(true);
-    expect(ghpages.options.repo)
-      .toEqual(`https://${process.env.GITHUB_TOKEN}@github.com/${process.env.TRAVIS_REPO_SLUG}.git`);
-    expect(ghpages.options.user).toEqual({ name: 'Deployment Bot', email: 'deploy@travis-ci.org' });
-  });
-
-  test('Site deploy -t/--travis deploys with custom GitHub repo', async () => {
-    process.env.TRAVIS = true;
-    process.env.GITHUB_TOKEN = 'githubToken';
-    process.env.TRAVIS_REPO_SLUG = 'TRAVIS_USER/TRAVIS_REPO.git';
-
-    const customRepoConfig = JSON.parse(SITE_JSON_DEFAULT);
-    customRepoConfig.deploy.repo = 'https://github.com/USER/REPO.git';
-    const json = {
-      ...PAGE_NJK,
-      'site.json': JSON.stringify(customRepoConfig),
-      _site: {},
-    };
-    fs.vol.fromJSON(json, '');
-    const site = new Site('./', '_site');
-    await site.deploy(true);
-    expect(ghpages.options.repo)
-      .toEqual(`https://${process.env.GITHUB_TOKEN}@github.com/USER/REPO.git`);
-  });
-
-  test('Site deploy -t/--travis deploys to correct repo when .git is in repo name', async () => {
-    process.env.TRAVIS = true;
-    process.env.GITHUB_TOKEN = 'githubToken';
-    process.env.TRAVIS_REPO_SLUG = 'TRAVIS_USER/TRAVIS_REPO.github.io';
-
-    const json = {
-      ...PAGE_NJK,
-      'site.json': SITE_JSON_DEFAULT,
-      _site: {},
-    };
-    fs.vol.fromJSON(json, '');
-    const site = new Site('./', '_site');
-    await site.deploy(true);
-    expect(ghpages.options.repo)
-      .toEqual(`https://${process.env.GITHUB_TOKEN}@github.com/TRAVIS_USER/TRAVIS_REPO.github.io.git`);
-  });
-
-  test('Site deploy -t/--travis should not deploy if not in Travis', async () => {
-    const json = {
-      ...PAGE_NJK,
-      'site.json': SITE_JSON_DEFAULT,
-      _site: {},
-    };
-    fs.vol.fromJSON(json, '');
-    const site = new Site('./', '_site');
-    await expect(site.deploy(true))
-      .rejects
-      .toThrow(new Error('-t/--travis should only be run in Travis CI.'));
-  });
-
-  test('Site deploy -t/--travis should not deploy without authentication token', async () => {
-    process.env.TRAVIS = true;
+const DEFAULT_GITHUB_TOKEN = 'githubToken';
+const DEFAULT_TRAVIS_REPO_SLUG = 'TRAVIS_USER/TRAVIS_REPO';
+const DEFAULT_APPVEYOR_REPO_SLUG = 'APPVEYOR_USER/APPVEYOR_REPO';
+const DEFAULT_TRAVIS_USER = {
+  name: 'Deployment Bot',
+  email: 'deploy@travis-ci.org',
+};
+const DEFAULT_APPVEYOR_USER = {
+  name: 'AppVeyorBot',
+  email: 'deploy@appveyor.com',
+};
+
+test.each(
+  [
+    // default settings
+    [
+      true,
+      DEFAULT_TRAVIS_REPO_SLUG,
+      false,
+      DEFAULT_APPVEYOR_REPO_SLUG,
+      DEFAULT_GITHUB_TOKEN,
+      false,
+      false,
+      true,
+      'https://githubToken@github.com/TRAVIS_USER/TRAVIS_REPO.git',
+      DEFAULT_TRAVIS_USER,
```

Comment:
Sorry if this was misinterpreted, this is definitely far too complex.

What I meant was to make each test into a matrix, not the entire test suite into one giant matrix.

For e.g. for the first test:

```
// Keep the overarching `describe`, rename into something more representative
describe('Site deploy with Travis', () => { // into describe('Site deploy with various CI environments', () => {

// then for e.g. 

test.each([
  ['TRAVIS', 'TRAVIS_REPO_SLUG'],
  ['APPVEYOR', 'APPVEYOR_REPO_NAME'],
])
  .('Site deploy -c/--ci deploys with default settings', async (identifierVar, slugVar) => {
    process.env[identifierVar]= true;
    process.env.GITHUB_TOKEN = 'githubToken';
    process.env[slugVar] = 'GENERIC_USER/GENERIC_REPO';

    const json = {
      ...PAGE_NJK,
      'site.json': SITE_JSON_DEFAULT,
      _site: {},
    };
    fs.vol.fromJSON(json, '');
    const site = new Site('./', '_site');
    await site.deploy(true);
    expect(ghpages.options.repo)
      .toEqual(`https://${process.env.GITHUB_TOKEN}@github.com/${process.env[slugVar]}.git`);
    expect(ghpages.options.user).toEqual({ name: 'Deployment Bot', email: 'deploy@travis-ci.org' });
  });
```

Some of the tests also don't need a matrix at all as well, (e.g. `Site deploy -c/--ci should not deploy if not in Travis`) as the test inputs are always the same

Reply:
should also remove your concerns 1 & 2 above entirely https://github.com/MarkBind/markbind/pull/1453#discussion_r562116768

Reply:
Ahh I see. Thanks for the clarification! Will make the changes.